integration/singularity/reader.go

```diff
@@ -0,0 +1,105 @@
+package singularity
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/data-preservation-programs/singularity/client/swagger/http/file"
+	"github.com/gotidy/ptr"
+)
+
+// io.ReadSeekCloser implementation that reads from remote singularity
+type SingularityReader struct {
+	store  *SingularityStore
+	fileID uint64
+	offset int64
+	size   int64
+}
+
+func (r *SingularityReader) Read(p []byte) (int, error) {
+	buf := bytes.NewBuffer(p)
+	buf.Reset()
+	writer := &signalWriter{
```

Comment:
I don't think you need the whole signalWriter

I believe RetrieveFile is a synchronous call, and at the end of it, the entire buffer is written.

Do you have experimental evidence that this is not the case? (i.e. retrieve file returns before the whole buffer is written?

Reply:
you're right, i did a test and there does not seem to be a sync issue, i'll take this out!